src/hookintrprtr.h

```diff
@@ -0,0 +1,178 @@
+/****************************************************************************
+**
+*W  hookintrprtr.h                    GAP source              Chris Jefferson
+**
+**
+*Y  Copyright (C) 2017 The GAP Group
+**
+**  This file contains functions related to hooking the interpreter.
+**  This code was abstracted out from the line-by-line profiling, for other
+**  uses.
```

Comment:
Instead of refering to history, why not be more proactive: Replace the last sentence by e.g. "These functions are used to implement line-by-line profiling, and a debugger." Or if you want to be defensive about the debugger: "These functions are used to implement line-by-line profiling, and are potentially useful for other appplications, such as a debugger."